File: artifacts/ai-business-os/src/services/activation-readiness.ts
import { apiClient, type ApiClient } from "./api-client.ts";

export type ActivationReadinessState =
  | "ready"
  | "action_needed"
  | "not_applicable";

export type ActivationReadinessCheck = {
  id: string;
  label: string;
  state: ActivationReadinessState;
  required: boolean;
  detail: string;
  href: string;
  evidence: Record<string, string | number | boolean | null>;
};

export type ActivationReadiness = {
  activation_ready: boolean;
  overall_status: "ready" | "action_needed";
  ready_required_checks: number;
  required_checks: number;
  checks: ActivationReadinessCheck[];
  generated_at: string;
};

export function createActivationReadinessApi(client: ApiClient) {
  return {
    get: (businessId: string, signal?: AbortSignal) =>
      client.request<ActivationReadiness>(
        `/api/v1/businesses/${encodeURIComponent(businessId)}/activation-readiness`,
        { signal },
      ),
  };
}

export const activationReadinessApi = createActivationReadinessApi(apiClient);

File: backend/app/schemas/activation_readiness.py
from __future__ import annotations

from typing import Literal

from pydantic import AwareDatetime, BaseModel, ConfigDict, Field


ReadinessState = Literal["ready", "action_needed", "not_applicable"]


class ActivationReadinessCheck(BaseModel):
    id: str = Field(pattern=r"^[a-z][a-z0-9_]{0,63}$")
    label: str = Field(min_length=1, max_length=120)
    state: ReadinessState
    required: bool
    detail: str = Field(min_length=1, max_length=500)
    href: str = Field(pattern=r"^/[A-Za-z0-9/_-]*$")
    evidence: dict[str, str | int | bool | None] = Field(
        default_factory=dict,
        max_length=12,
    )
    model_config = ConfigDict(extra="forbid")


class ActivationReadinessResponse(BaseModel):
    activation_ready: bool
    overall_status: Literal["ready", "action_needed"]
    ready_required_checks: int = Field(ge=0)
    required_checks: int = Field(ge=1)
    checks: list[ActivationReadinessCheck] = Field(min_length=1, max_length=32)
    generated_at: AwareDatetime
    model_config = ConfigDict(extra="forbid")
